MySQLdb/_mysql.c

```diff
@@ -446,6 +446,8 @@ _mysql_ConnectionObject_Initialize(
 
     Py_BEGIN_ALLOW_THREADS ;
     conn = mysql_init(&(self->connection));
+    if (!conn)
+        return PyErr_NoMemory();
```

Comment:
Move Py_BEGIN_ALLOW_THREADS to after this line.

Reply:
Ah yeah, we moved Py_BEGIN_ALLOW_THREADS down to just before the connect, sorry missed this